### PR TITLE
lxd init: support all storage drivers

### PR DESCRIPTION
**Pull request by brauner**

### CODE REVIEW

**Comment by brauner** [comment]
@stgraber, fyi, the golang-sweep-increased-allocation-count fix got merged.

File: lxd/main_init.go [hunk 1]
```diff
@@ -417,18 +421,56 @@ they otherwise would.
...
+
+						// Unconditionally overwrite
+						// because if the user ends up
+						// with a clean LXD but with a
+						// pool property key existing in
+						// the default profile it must
+						// be empty otherwise it would
+						// not have been possible to
+						// delete the storage pool in
+						// the first place.
```
Comment (stgraber) [inline lxd/main_init.go]: We don't stick to 80 cols in the LXD code, you can wrap that comment to be a bit more readable :)

File: lxd/main_init.go [hunk 2]
```diff
@@ -249,9 +253,10 @@ func cmdInit() error {
 			}
 		}
 
-		if storageSetup && storageBackend == "zfs" {
+		if storageSetup && storageBackend != "dir" {
```
Comment (stgraber) [inline lxd/main_init.go]: So what's happening with dir exactly? Does the user just not get asked for pool name, ... if it's a dir backend?

If so, does that mean that they can't use "lxd init" to define a second dir pool?

File: lxd/main_init.go [hunk 3]
```diff
@@ -40,17 +39,20 @@ func cmdInit() error {
...
 
 	backendsAvailable := []string{"dir"}
-	backendsSupported := []string{"dir", "zfs"}
 
-	// Detect zfs
-	out, err := exec.LookPath("zfs")
-	if err == nil && len(out) != 0 && !runningInUserns {
-		_ = loadModule("zfs")
+	// Check available backends
+	for _, driver := range supportedStoragePoolDrivers {
```
Comment (brauner) [inline lxd/main_init.go]: Note to self: Needs to account for running in userns where only dir needs to be made available.

Comment (stgraber) [inline lxd/main_init.go]: Some level of btrfs can work too (only using source=/some/path/on/btrfs).

File: lxd/main_init.go (hunk 2 above)
Comment (brauner) [inline lxd/main_init.go]: Hm, why? I just created two dir pools with two lxd init runs named `bla` and `blu`. This condition only excludes block devices and loop files for dir.

File: lxd/main_init.go (hunk 1 above)
Comment (brauner) [inline lxd/main_init.go]: Will do. :)

File: lxd/main_init.go (hunk 2 above)
Comment (stgraber) [inline lxd/main_init.go]: Ah, makes sense, sorry was missing context in this review.

File: lxd/main_init.go (hunk 3 above)
Comment (brauner) [inline lxd/main_init.go]: Yup, accounted for that.

**Pull request merged**